### PR TITLE
fix(api): add tags to the parameter wrapper in article api endpoints

### LINKED ISSUE #11516: Issue in POST /articles api endpoint: tags are not returned in the initial creation response
**Describe the bug**

I'm using API to create an article. I'm sending the tags as a list of strings in the request payload. But, can't find the tags in the response or in the articles on dev.to. 

**To Reproduce**

1. Send an API request to /articles for creating a article. 
For eg.
```
.{
  title: 'Quickwork Testing',
  body_markdown: '# asdasd',
  published: false,
  series: undefined,
  main_image: 'https://avatars3.githubusercontent.com/u/36989112?s=460&v=4',
  canonical_url: 'https://dasdasda.com',
  description: 'dsafadfsd',
  tags: [ 'javascript' ],
  organization_id: NaN
}

```

2. This the response received.

```
{
  "type_of": "article",
  "id": 520013,
  "title": "Quickwork Testing",
  "description": "dsafadfsd",
  "readable_publish_date": "Nov 20",
  "slug": "quickwork-testing-ape",
  "path": "/nishantwrp/quickwork-testing-ape",
  "url": "https://dev.to/nishantwrp/quickwork-testing-ape",
  "comments_count": 0,
  "public_reactions_count": 0,
  "published_timestamp": "2020-11-20T11:05:48Z",
  "positive_reactions_count": 0,
  "cover_image": "https://res.cloudinary.com/practicaldev/image/fetch/s--q96skcRN--/c_imagga_scale,f_auto,fl_progressive,h_420,q_auto,w_1000/https://avatars3.githubusercontent.com/u/36989112%3Fs%3D460%26v%3D4",                                                                                                                                                                      
  "social_image": "https://res.cloudinary.com/practicaldev/image/fetch/s--184jV2TS--/c_imagga_scale,f_auto,fl_progressive,h_500,q_auto,w_1000/https://avatars3.githubusercontent.com/u/36989112%3Fs%3D460%26v%3D4",                                                                                                                                                                     
  "canonical_url": "https://dasdasda.com",
  "created_at": "2020-11-20T11:05:48Z",
  "published_at": "2020-11-20T11:05:48Z",
  "last_comment_at": "2020-11-20T11:05:48Z",
  "tag_list": "",
  "tags": [],
  "body_html": "<h1>\n  <a name=\"asdasd\" href=\"#asdasd\" class=\"anchor\">\n  </a>\n  asdasd\n</h1>\n\n",
  "body_markdown": "# asdasd",
  "user": {
    "name": "Nishant Mittal",
    "username": "nishantwrp",
    "twitter_username": "nishantwrp",
    "github_username": "nishantwrp",
    "website_url": "https://www.nishantwrp.com",
    "profile_image": "https://res.cloudinary.com/practicaldev/image/fetch/s--UUPQ1U-G--/c_fill,f_auto,fl_progressive,h_640,q_auto,w_640/https://dev-to-uploads.s3.amazonaws.com/uploads/user/profile_image/239671/98b595a2-72f3-4814-9e6d-b14ede18cfff.jpeg",
    "profile_image_90": "https://res.cloudinary.com/practicaldev/image/fetch/s--ffdeoH58--/c_fill,f_auto,fl_progressive,h_90,q_auto,w_90/https://dev-to-uploads.s3.amazonaws.com/uploads/user/profile_image/239671/98b595a2-72f3-4814-9e6d-b14ede18cfff.jpeg"
  }
}

```

**Expected behavior**

It can be seen that the tags property is empty array but shouldn't be.

**Screenshots**

**Desktop (please complete the following information):**

- OS, version:
- Browser, version:

**Smartphone (please complete the following information):**

- Device:
- OS, version:
- Browser, version:

**Additional context**

### PR DESCRIPTION
**Pull request by nishantwrp**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

### The issue
When I try to create an article using the `/articles` endpoint with the following payload.
```
{
        "title": "A test post again",
        "description": "Testing article creation bug again",
        "body_markdown": "## Some content",
        "tags": ["javascript"]
}
```

The article gets created but there are no tags in it.

However when I access the same endpoint using the following payload. 
```
{
  article: {
        "title": "A test post again",
        "description": "Testing article creation bug again",
        "body_markdown": "## Some content",
        "tags": ["javascript"]
  }
}
```

There is no such issue.

### The cause
Btw, according to documentation, there should be an `article` property in which all the article properties are present. But not adding the `article` property doesn't result in an error in the response but there is a little bug as mentioned above. 

This is because of the parameter wrapper by rails defined here.
https://github.com/forem/forem/blob/65110550d8c2231f73ff99ae305a46b7b0bcb4f9/config/initializers/wrap_parameters.rb#L7-L9

When a request is received to the articles endpoint without the `article` property it automatically creates one and the final `params` variable has the following value.
```
{"title"=>"A test post again", "description"=>"Testing article creation bug again", "body_markdown"=>"## Some content", "tags"=>["javascript"], "locale"=>nil, "article"=>{"title"=>"A test post again", "body_markdown"=>"## Some content", "description"=>"Testing article creation bug again"}}
```

It can be seen that it wrapped everything except the `tags` probably because it's an array.

### The solution implemented
So, I manually defined the parameters that should be wrapped in the `ArticlesController`.

### Alternative solution
If not this, we can disable the wrap_parameters globally (or maybe in this controller only) too, so that there's an error when the user doesn't wrap the article body in `article` property in the post/put request.

## Related Tickets & Documents
Fixes #11516 

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how accessibility is impacted and tested. For more info, check out the [Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by nishantwrp** [comment]
@rhymes ptal!

**Comment by rhymes** [comment]
@nishantwrp could you add a regression spec for that? Making sure that the first version now actually creates an article with those tags. Thank you!

**Comment by nishantwrp** [comment]
@rhymes done.

**Comment by nishantwrp** [comment]
Btw, I was just wondering that similar issues could be present in other endpoints too. (Non - `articles` endpoints)

**Comment by rhymes** [comment]
@nishantwrp I'm not sure, I think it would be better in general to rely on the documentation and possibly return an error if the payload is malformed but for the time being this is probably fine as it is.

File: app/controllers/api/v0/articles_controller.rb [hunk 1]
```diff
@@ -11,6 +11,11 @@ class ArticlesController < ApiController
 
       skip_before_action :verify_authenticity_token, only: %i[create update]
 
+      wrap_parameters :article, include: %i[
+        title body_markdown published series main_image
+        canonical_url description tags
+      ]
```
Comment (rhymes) [inline app/controllers/api/v0/articles_controller.rb:17]: I just realized that this list is repeated down in `article_params` and it doesn't include `organization_id`

**Review by rhymes — COMMENTED** [review]
I'm still a bit conflicted about this PR because it's a work around on a problem we shouldn't have since the documentation states that the payload should be inside `{"article"}`. How Rails treats params is a bit counterintuitive, what if instead we disable parameters wrapping for this controller, with `wrap_parameters: false`

**Comment by nishantwrp** [comment]
@rhymes I agree with you, maybe we can globally disable this wrapping of parameters?

**Comment by rhymes** [comment]
I made a note of it in https://github.com/forem/forem/issues/11645 - will investigate further ASAP

**Comment by nishantwrp** [comment]
Cool, should i close this pr then, or should I temporarily disable parameter wrapping in this pr, then?

**Comment by rhymes** [comment]
@nishantwrp would it be okay to keep it here open until we understand the consequences of #11645 better?

**Comment by nishantwrp** [comment]
Yup, sure.

**Comment by rhymes** [comment]
@nishantwrp as we disabled parameter's wrapping in https://github.com/forem/forem/pull/11692 I'm going ahead and close this. Thank you for the nudge that lead us to the fix!

**Pull request closed without merging**